src/cogs/leveling.py

```diff
@@ -0,0 +1,82 @@
+import discord
+import random
+import json
+import datetime
+
+from ..helpers.cooldownmgr import CooldownManager
+from ..util.levels_xp import levels
+from discord.ext import commands
+
+
+class Leveling(commands.Cog):
+
+    def __init__(self, client):
+        self.bot = client
+        self.pool = client.pool
+        self.cdmanager = CooldownManager(client.loop)
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message):
+        con = await self.pool.acquire()
+
+        author_id = message.author.id
+        user_query = """SELECT * FROM users WHERE discord_id=$1"""
+        user_info = await con.fetchrow(user_query, author_id)
+
+        if not user_info:
+            create_user_query = """INSERT INTO users (discord_id, username, level, xp) VALUES ($1, $2, $3, $4) ON CONFLICT DO NOTHING"""
+            await con.execute(create_user_query, author_id,
+                              message.author.name, 0, 0)
+        else:
+            if self.cdmanager.on_cooldown(author_id):
+                return
+
+            fianl_xp = user_info['xp'] + random.randint(15, 30)
+            final_level = user_info['level']
+
+            if fianl_xp > levels[user_info['level']+1][0]:
+                print('level up')
+                final_level += 1
+                if levels[final_level+1][1] is not None:
```

Comment:
this might raise a KeyError if one is at level 100, but I think noone got there yet so should be fine lol